### ISSUE TITLE
Error running in yarn due to undeclared dependency

### ISSUE THREAD

**Issue by dobesv** [happo/happo.io#268]
Yarn is strict about using dependencies that are not declared in package.json - if a file imports or requires a module it must be declared in that file's package.json.  This differs from legacy environments where a module could use the dependencies of its dependencies freely.

```
 yarn run happo dev
Starting: [main] Searching for happo test files... 
✗ [main] Searching for happo test files...: Error: happo.io tried to access react-dom, but it isn't declared in its dependencies; this makes the require call ambiguous and unsound.

Required package: react-dom
Required by: happo.io@virtual:0b8541be94c07b03e28e04a869f424dd9b9352f559e63e47625f947ec9ee14213658be2582b051a033b49b2f2beeedb9f8fde576b107aac3aa921ebdc44b4415#npm:9.1.2

```

**Comment by ljharb** [comment]
note that it's not "legacy" environments, it's "the standard environment" (ie, npm).

**Comment by dobesv** [comment]
Yeah it can be worked around changing `.yarnrc` but obviously that won't be friendly to new users.

There are two resolutions to this issue that I know of:

1. Add an optional `peerDependency` - e.g. add it to `peerDependencies` and also mark it optional in `peerDependenciesMeta`.  This allows you to use it if you want to withoutwarnings if it is not present. I'm not 100% if/how npm supports this one though.
2. Use `createRequire` from https://nodejs.org/api/module.html#modulecreaterequirefilename to dynamically import the module from within the target package's import context, in this case it is treated as if you are a file in that target package for purposes of module resolution

Which you choose depends on the details of what you are up to.  My guess is that #1 should work OK.

**Comment by ljharb** [comment]
It depends on what node/npm versions you support; npm certainly supports peerDependenciesMeta, but versions of npm prior to that will treat the peer dep as required.
